front_end/content.py

```diff
@@ -8,310 +8,490 @@
-    problem_dict = {"id": problem, "title": "", "visible": True, "exists": False, "assignment": assignment_basics}
-
-    if os.path.exists(file_path):
-        problem_dict = load_yaml_dict(read_file(file_path))
-        problem_dict["exists"] = True
-        problem_dict["assignment"] = assignment_basics
-
-    return problem_dict
-
-def get_next_prev_problems(course, assignment, problem, problems):
-    prev_problem = None
-    next_problem = None
-
-    if len(problems) > 0 and problem:
-        this_problem = [i for i in range(len(problems)) if problems[i][0] == problem]
-        if len(this_problem) > 0:
-            this_problem_index = [i for i in range(len(problems)) if problems[i][0] == problem][0]
-
-            if len(problems) >= 2 and this_problem_index != 0:
-                prev_problem = problems[this_problem_index - 1][1]
-
-            if len(problems) >= 2 and this_problem_index != (len(problems) - 1):
-                next_problem = problems[this_problem_index + 1][1]
-
-    return {"previous": prev_problem, "next": next_problem}
-
-def get_next_submission_id(course, assignment, problem, user):
-    return get_num_submissions(course, assignment, problem, user) + 1
-
-def get_last_submission(course, assignment, problem, user):
-    last_submission_id = get_num_submissions(course, assignment, problem, user)
-    file_path = f"/submissions/{course}/{assignment}/{problem}/{user}/{last_submission_id}.yaml"
-    last_submission = load_yaml_dict(read_file(file_path))
-
-    return last_submission
-
-def get_num_submissions(course, assignment, problem, user):
-    return len(glob.glob(f"{get_submission_dir_path(course, assignment, problem, user)}"))
-
-def get_submission_info(course, assignment, problem, user, submission_id):
-    return load_yaml_dict(read_file(get_submission_file_path(course, assignment, problem, user, submission_id)))
-
-def get_course_details(course, format_output=False):
-    file_path = get_course_dir_path(course) + "details"
-    exists = os.path.exists(file_path)
-    course_dict = {"introduction": ""}
-
-    if exists:
-        course_dict = load_yaml_dict(read_file(file_path))
-        if format_output:
-            course_dict["introduction"] = convert_markdown_to_html(course_dict["introduction"])
-
-    return course_dict
-
-def get_assignment_details(course, assignment, format_output=False):
-    file_path = get_assignment_dir_path(course, assignment) + "details"
-    exists = os.path.exists(file_path)
-    assignment_dict = {"introduction": ""}
-
-    if exists:
-        assignment_dict = load_yaml_dict(read_file(file_path))
-        if format_output:
-            assignment_dict["introduction"] = convert_markdown_to_html(assignment_dict["introduction"])
-
-    return assignment_dict
-
-def get_problem_details(course, assignment, problem, format_content=False, parse_data_urls=False):
-    file_path = get_problem_dir_path(course, assignment, problem) + "details"
+class Content:
```

Comment:
This class looks good. Please update the code in front_end/summarize_logs.py to use the new version of content.py.